[pytket-qiskit] Test failure: `test_ibmq_mid_measure`.

This test requires IBMQ credentials so isn't run on the CI.
It fails with the following error:
```
____________________________________________________ test_ibmq_mid_measure _____________________________________________________

    @pytest.mark.skipif(skip_remote_tests, reason="Skipping remote tests")
    def test_ibmq_mid_measure() -> None:
        c = Circuit(3, 3).H(1).CX(1, 2).Measure(0, 0).Measure(1, 1)
        c.add_barrier([0, 1, 2])
    
        c.CX(1, 0).H(0).Measure(2, 2)
    
        # test open backend does not support mid-measure
        # cannot test premium backends that do
        b = IBMQEmulatorBackend("ibmq_athens")
    
        with pytest.raises(RuntimeError) as warninfo:
            b.compile_circuit(c)
>           assert "Not a valid operation" in str(warninfo.value)

backend_test.py:829: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <ExceptionInfo for raises contextmanager>

    @property
    def value(self) -> _E:
        """The exception value."""
>       assert (
            self._excinfo is not None
        ), ".value can only be used after the context manager exits"
E       AssertionError: .value can only be used after the context manager exits
```

--- a/modules/pytket-qiskit/tests/backend_test.py
+++ b/modules/pytket-qiskit/tests/backend_test.py
@@ -24,7 +24,7 @@ import numpy as np
 from pytket.circuit import Circuit, OpType, BasisOrder, Qubit, reg_eq  # type: ignore
 from pytket.passes import CliffordSimp  # type: ignore
 from pytket.pauli import Pauli, QubitPauliString  # type: ignore
-from pytket.predicates import CompilationUnit  # type: ignore
+from pytket.predicates import CompilationUnit, NoMidMeasurePredicate  # type: ignore
 from pytket.routing import Architecture, route  # type: ignore
 from pytket.device import Device  # type: ignore
 from pytket.transform import Transform  # type: ignore
@@ -798,17 +798,10 @@ def test_ibmq_mid_measure() -> None:
 
     c.CX(1, 0).H(0).Measure(2, 2)
 
-    # test open backend does not support mid-measure
-    # cannot test premium backends that do
     b = IBMQEmulatorBackend("ibmq_athens")
-
-    with pytest.raises(RuntimeError) as warninfo:
-        b.compile_circuit(c)
-        assert "Not a valid operation" in str(warninfo.value)
-
-    with pytest.raises(CircuitNotValidError) as warninfo2:
-        b.get_counts(c, 10)
-        assert "NoMidMeasurePredicate" in str(warninfo2.value)
+    b.compile_circuit(c)
+    assert not NoMidMeasurePredicate().verify(c)
+    assert b.valid_circuit(c)
 
 
 @pytest.mark.skipif(not IBMQ.stored_account(), reason="No IBM account stored")

